docker/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:16.04
+MAINTAINER koalaman <koalaman@gmail.com>
+
+RUN apt-get update -qy
+
+RUN apt-get -qy install git cabal-install
+
+RUN cd $HOME && git clone https://github.com/koalaman/shellcheck.git
+RUN cd $HOME/shellcheck && cabal update
+RUN cd $HOME/shellcheck && cabal install
+
+
+# Clean apt cache
+RUN apt-get clean && \
```

Comment:
You need to run these cleanup tasks in the same RUN statement as your update/install.  Otherwise you'll have a layer with all of these files, and then another layer that removes them.  You won't be getting the space savings that you expect.